src/server.rs

```diff
@@ -355,7 +361,7 @@ async fn stream_classification_with_gen(
             .boxed(),
         );
     }
-    let task = StreamingClassificationWithGenTask::new(request_id, request);
+    let task = StreamingClassificationWithGenTask::new(request_id, request, headers);
```

Comment:
Update per other changes and apply `filter_headers()`

```suggestion
    let headers = filter_headers(&state.orchestrator.config().passthrough_headers, headers);
    let task = StreamingClassificationWithGenTask::new(request_id, request, headers);
```